<!DOCTYPE html>
<html>

<body>
  <h1>Practice Problems Webpage</h1>
  <p>Your solutions for problems will print in the Replit console.</p>
  <p>The 404 error you see in the console is not a crash, and is expected.</p>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/axios/1.4.0/axios.min.js"></script>
  <script>
    (async function () { // Immediately-Invoked Function Expression (IIFE) with async keyword
      /*
      PROBLEM 1:
      Finish converting the request for Pluto into the await syntax by completing the try block.
      The await keyword already exists in the IIFE that wraps the whole script (see line 10).
      */
      const pluto = 'https://webapis.bloomtechdev.com/planets/9'
      axios.get(pluto)
        .then(res => { console.log(`Pluto is the ${res.data.id}th planet!`) })
        .catch(err => { console.log(`Something went wrong: ${err.message}`) })

      try {
        // ENTER YOUR CODE HERE
      } catch (err) {
        console.log(`Something went wrong: ${err.message}`)
      }

      /*
      PROBLEM 2:
      Finish converting the request for a non-existing planet into the await syntax by completing the try/catch blocks.
      */
      const badPlanet = 'https://webapis.bloomtechdev.com/planets/12345'
      axios.get(badPlanet)
        .then(res => { console.log(`${res.data.name} is in your dreams!`) })
        .catch(err => { console.log(`Something went wrong: ${err.message}`) })

      try {
        // ENTER CODE HERE
      } catch (err) {
        // ENTER MORE CODE HERE
      }

      /*
      PROBLEM 3:
      Recreate the functionality below using try/catch blocks and the await syntax.
      */
      const planetList = 'https://webapis.bloomtechdev.com/planets'
      axios.get(planetList)
        .then(res => {
          const planets = res.data
          const legitPlanets = planets.filter(planet => planet.name !== 'Pluto')
          console.log('The legit planets are:')
          legitPlanets.forEach(planet => {
            console.log(`\t* ${planet.name}`)
          })
        })
        .catch(err => { console.log(err.message) })
      // ENTER YOUR CODE HERE

    }())
  </script>
</body>

</html>
